### PR TITLE
feat: Add enableNetworkTracking flag

### PR DESCRIPTION
**Pull request by philipphofmann**
## :scroll: Description

Add enableNetworkTracking to SentryOptions for disabling the
SentryNetworkTrackingIntegration easily.

## :bulb: Motivation and Context

Fixes GH-1327

## :pencil: Checklist

- [x] I reviewed the submitted code
- [x] I added tests to verify the changes
- [x] I updated the docs if needed
- [x] Review from the native team if needed
- [x] No breaking changes

### CODE REVIEW

File: Sources/Sentry/SentryNetworkTrackingIntegration.m [hunk 1]
```diff
@@ -1,22 +1,31 @@
...
 @implementation SentryNetworkTrackingIntegration
 
 - (void)installWithOptions:(SentryOptions *)options
 {
-    self.options = options;
-    if (options.enableAutoPerformanceTracking) {
-        [SentryNetworkSwizzling start];
+    // We don't check isTracingEnabled, because the integration
+    // also creates breadcrumbs for HTTP requests.
+    if (!options.enableAutoPerformanceTracking) {
```
Comment (brustolin) [inline Sources/Sentry/SentryNetworkTrackingIntegration.m:13]: l: And in the case that I want only to track network request but not UI performance?
I would prefer this two flags to be independents.

**Review by brustolin — APPROVED** [review]
Everything looks good. I made only one suggestion about the flags.

File: Sources/Sentry/SentryNetworkTrackingIntegration.m (hunk 1 above)
Comment (philipphofmann) [reply Sources/Sentry/SentryNetworkTrackingIntegration.m:13]: Good point. This should actually be `h` 😁. Will change it.

Comment (philipphofmann) [reply Sources/Sentry/SentryNetworkTrackingIntegration.m:13]: I guess you still want HTTP breadcrumbs when you disable `enableAutoPerformanceTracking`. I'm going to change it. This should be `h` 😁

Comment (philipphofmann) [reply Sources/Sentry/SentryNetworkTrackingIntegration.m:26]: I changed my mind. I'm not going to fix this right now. I put a comment into this class

> We are aware that the SDK only creates breadcrumbs for HTTP requests if performance is
> enabled. A proper fix is not straightforward as we need several checks on SentryOptions in
> SentryNetworkTracker. As we have a problem with KVO, see
> https://github.com/getsentry/sentry-cocoa/issues/1328, we don't know if we can keep the
> SentryNetworkTracker (written on 29th of September 2021). Therefore we accept this tradeof
> for now.

**Review by brustolin — APPROVED** [review]
I agree with you. Let's review the flags once we have the network tracking figure it out.

**Pull request merged**